### PR TITLE
Remove exposing windows.h header from our public headers within storage.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Follow up to https://github.com/Azure/azure-sdk-for-cpp/pull/2188, fixes https://github.com/Azure/azure-sdk-for-cpp/issues/2182

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]

**Comment by ahsonkhan** [comment]
@Jinming-Hu I am waiting for your review before merging this. Should be fairly quick :)

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]
This is a .cpp file, why do we need to remove `windows.h` from .cpp file?

**Comment by ahsonkhan** [comment]
> This is a .cpp file, why do we need to remove `windows.h` from .cpp file?

We don't, fixed the PR to update the header file instead. Good catch.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**
